File: hotel_name_autocomplete/v1/get/Python SDK/hotel_name_autocomplete.py
# Install the Python library from https://pypi.org/project/amadeus
from ast import keyword
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

try:
    '''
    Hotel name autocomplete for keyword 'PARI' using HOTEL_GDS category of search
    '''
    response = amadeus.reference_data.locations.hotel.get(keyword='PARI', subType=[Hotel.HOTEL_GDS])
    print(response.data)
except ResponseError as error:
    raise error

File: hotel_name_autocomplete/v1/get/Node SDK/hotel_name_autocomplete.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});
// Or `const amadeus = new Amadeus()` if the environment variables are set

async function main() {
  try {
    // Hotel name autocomplete for keyword 'PARI' using  HOTEL_GDS category of search
    const response = await amadeus.referenceData.locations.hotel.get({
      keyword: "PARI",
      subType: "HOTEL_GDS",
    });

    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

main();
